pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1400,6 +1389,9 @@ func (c *PodAffinityChecker) nodeMatchesTopologyTerms(pod *v1.Pod, topologyPairs
 				return false
 			}
 		} else {
+			// TODO(Huang-Wei): if incoming pod has an anti-affinity with invalid topology
```

Comment:
We should have two functions, instead of this one: `nodeMatchesAllTopologyTerms` and `nodeMatchesAnyTopologyTerms`. We should then use `nodeMatchesAllTopologyTerms` for checking affinity and `nodeMatchesAnyTopologyTerms` for checking anti-affinity.

Reply:
Good point. Done.